tools/waveica/waveica_wrapper.R

```diff
@@ -1,4 +1,56 @@
-waveica <- function(data,
+read_file <- function(file, metadata, ft_ext, mt_ext) {
+  data <- read_ft_data(file, ft_ext)
+
+  if (!is.na(metadata)) {
+    mt_data <- read_metadata(metadata, mt_ext)
+    data <- merge(mt_data, data, by = "sampleName")
+  }
+
+  return(data)
+}
+
+read_ft_data <- function(file, ft_ext) {
+  if (ft_ext == "csv") {
+    ft_data <- read.csv(file, header = TRUE)
+  } else if (ft_ext == "tsv") {
+    ft_data <- read.csv(file, header = TRUE, sep = "\t")
+  } else {
+    ft_data <- arrow::read_parquet(file)
+  }
+
+  return(ft_data)
+}
+
+read_metadata <- function(metadata, mt_ext) {
+  if (mt_ext == "csv") {
+    mt_data <- read.csv(metadata, header = TRUE)
+  } else if (mt_ext == "tsv") {
+    mt_data <- read.csv(metadata, header = TRUE, sep = "\t")
+  } else {
+    mt_data <- arrow::read_parquet(metadata)
+  }
+
+  return(mt_data)
+}
+
+write_csv <- function(data, output) {
+  write.csv(data, file = output, row.names = FALSE, quote = FALSE)
+  cat("Normalization has been completed.\n")
+}
+
+write_tsv <- function(data, output) {
+  write.table(data, file = output, sep = "\t", row.names = FALSE, quote = FALSE)
+  cat("Normalization has been completed.\n")
+}
+
+write_parquet_file <- function(data, output) {
+  arrow::write_parquet(data, sink = output)
+  cat("Normalization has been completed.\n")
+}
```

Comment:
Same here - this could be one function, using the "flavor" as an argument - that would also make the wrapper a lot easier.